Added new kb article grid-kb-loader-during-contextmenu-actions (#2717)
telerik/blazor-docs: knowledge-base/grid-loader-during-contextmenu-actions.md

---

> title: Display Loader During Actions in TelerikGrid with ContextMenu

Reviewer: ```suggestion
title: Display Loader During ContextMenu Actions in Grid
```

> description: Learn how to show a loading indicator while performing actions from a ContextMenu in a TelerikGrid for Blazor.

Reviewer: ```suggestion
description: Learn how to show a loading indicator while performing actions from a ContextMenu in a Telerik Grid for Blazor.
```

type: how-to
page_title: How to Display a Loader in TelerikGrid for Blazor During ContextMenu Actions
slug: grid-kb-loader-during-contextmenu-actions

> tags: contextmenu,grid,loader,loading, loadercontainer,actions

Reviewer: ```suggestion
tags: contextmenu, grid, loader, loading, loadercontainer
```

    private IEnumerable<SampleData> SelectedItems { get; set; } = Enumerable.Empty<SampleData>();
    private SampleData SelectedPerson { get; set; }
    private TelerikContextMenu<MenuItem> ContextMenuRef { get; set; }
    private TelerikGrid<SampleData> GridRef { get; set; }
    private bool LoaderVisible { get; set; }

    public class MenuItem
    {
        public string Text { get; set; }
        public ISvgIcon Icon { get; set; }
        public Action Action { get; set; }
        public string CommandName { get; set; }
    }

    private async Task OnContextMenu(GridRowClickEventArgs args)
    {
        var argsItem = args.Item as SampleData;

        SelectedPerson = argsItem;

        if (args.EventArgs is MouseEventArgs mouseEventArgs)
        {
            await ContextMenuRef.ShowAsync(mouseEventArgs.ClientX, mouseEventArgs.ClientY);
        }
    }

    private async Task ContextMenuClickHandler(MenuItem item)
    {
        if (item.Action != null)
        {
            item.Action.Invoke();
        }
        else
        {
            switch (item.CommandName)
            {
                case "BeginEdit":
                    LoaderVisible = true;

>                     await Task.Delay(3000); //initiates delay

Reviewer: This implementation may backfire at you, because the developer will replace `await Task.Delay(3000)` with their actual long task and the LoaderContainer will never show. Instead use something like this:

```suggestion
                    await Task.Delay(1); // triggers UI refresh for the LoaderContainer to show
                    await Task.Delay(3000); // replace with the actual long-running task
```

                    LoaderVisible = true;
                    await Task.Delay(3000); //initiates delay
                    LoaderVisible = false;

                    var selItems = SelectedItems.ToList();
                    if (SelectedItems.Contains(SelectedPerson))
                    {
                        selItems.Remove(SelectedPerson);
                    }
                    else
                    {
                        selItems.Add(SelectedPerson);
                    }
                    SelectedItems = selItems;
                    break;
                default:
                    break;
            }
        }
        SelectedPerson = null; // clean up
    }

    protected override void OnInitialized()
    {
        MenuItems = new List<MenuItem>()
        {
            new MenuItem(){ Text = "Select", Icon = SvgIcon.CheckboxChecked, CommandName="ToggleSelect" },
            new MenuItem(){ Text = "Edit", Icon = SvgIcon.Pencil, CommandName="BeginEdit" },
            new MenuItem(){ Text = "Delete", Icon = SvgIcon.Trash, Action = DeleteItem }
        };

        GridData = new ObservableCollection<SampleData>();
        var rand = new Random();

>         for (int i = 0; i < 100; i++)

Reviewer: Do not use zero ids
